### PR TITLE
USWDS - Checkbox and radio: Update tile color definitions

### LINKED ISSUE #5490: USWDS - Radio and checkbox: Revert tile border colors
## Summary
Release 3.5.0 inadvertently changed the border color for the Radio and Checkbox tile outlines. We should revert these styles back to the 3.4.0 styles. 

## Details

The tile border colors are defined with disabled color tokens. These settings received an update in 3.5.0, which made the border color darker. Note the color shifts between [3.4.0](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/release-3.4.0/?path=/story/components-form-inputs-radio--radio-tile) and [3.5.0](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/release-3.5.0/?path=/story/components-form-inputs-radio--radio-tile). This could play a role in the visual noise. 

image

This is the first part of a bigger discussion related to component styles and visual processing disorder in:  https://github.com/uswds/uswds/issues/5487

### To-do
- [ ] Investigate if we should be defining this color with disabled settings

### PR DESCRIPTION
**Pull request by amyleadem**
# Summary

**Updated radio and checkbox tile styling to have lighter borders.**

This change reverts border color styles to be more similar to what they were in 3.4.0 and _could_ help with comprehension for users with visual processing disorders. 

> **Note**
> This is the first part of the effort to make checkbox and radio tile styles more accessible to users with visual processing disorders. We will need to dive deeper into what we can do to reduce visual noise so that the content is easier to focus on. Issue #5487 tracks the larger effort for this. 

## Breaking change

This is not a breaking change.

## Related issue

Closes #5490

## Related pull requests

[Changelog PR](https://github.com/uswds/uswds-site/pull/2270)

## Preview link

- [Radio tile](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/al-radio-tile-color/?path=/story/components-form-inputs-radio--radio-tile)
- [Checkbox tile](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/al-radio-tile-color/?path=/story/components-form-inputs-checkbox--disabled-tile)
- [Disabled form inputs](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/al-radio-tile-color/?path=/story/patterns-forms--disabled-form-elements)

## Problem statement
In release 3.5.0, we updated disabled styles for our form elements, which inadvertently darkened the color of the borders on the checkbox and radio tile boxes. This color changed increased the visual noise, which can have a negative impact on users with visual processing disorder.

Reference: [Color definition change in 3.5.0](https://github.com/uswds/uswds/pull/5063/files#diff-b6aedd118f192b4c3488b079a0e2bd509dbff00384cd41aa894430acb9fecdeaR35-R36)

## Solution
The tile borders are not actually representing disabled states, so we can make the component more resilient by defining the border colors with system tokens instead.

The disabled states will continue to use disabled color tokens ("disabled"/"gray-50") and will now meet color contrast requirements.

image

## Testing and review
- Confirm that tile border styles more closely align with those found in [release-3.4.0](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/release-3.4.0/?path=/story/components-form-inputs-radio--radio-tile)
    - Note: Disabled states will look darker than in 3.4.0. This is expected. Disabled styles should meet contrast standards and also align with other current disabled form inputs. 
- Confirm that the switch from disabled state colors to system colors make sense. 
- Confirm no unexpected visual regressions

### CODE REVIEW

**Review by mejiaj — APPROVED** [review]
Thanks for updating. I tested:
- [x] Default checkbox & radio.
- [x] Checkbox & radio with tests uncommented.

File: packages/uswds-core/src/styles/mixins/helpers/checkbox-and-radio-colors.scss [hunk 1]
```diff
@@ -32,8 +32,8 @@
     1
   );
   $input-border-alpha: -0;
-  $color-input-disabled: color("disabled");
-  $color-input-disabled--weak: color("disabled-lighter");
+  $tile-border-color: color("gray-20");
```
Comment (mahoneycm) [inline packages/uswds-core/src/styles/mixins/helpers/checkbox-and-radio-colors.scss:35]: Any thoughts on making the border a bit darker at `gray-30`? 

Adds an extra step between the actual disabled element element and the default which might help to distinguish the difference between the two for lower visioned users

Disabled comparison

Comment (amyleadem) [reply packages/uswds-core/src/styles/mixins/helpers/checkbox-and-radio-colors.scss:35]: I lean towards keeping the border color "gray-20". My reasoning is that I don't think that the border color should do much work to indicate that the element is disabled. Instead, it should be the lightened text and radio button color that carries that lift. 

In my mind, this PR primarily exists to restore the styles from 3.4.0, which are more closely aligned with "gray-20". Secondarily, the aim is to reduce visual noise to make this easier to read for users with visual processing disorders. Lightening the borders is likely not enough for that effort, but it is a step in the right direction. 

Curious what you and others think!

Comment (mahoneycm) [reply packages/uswds-core/src/styles/mixins/helpers/checkbox-and-radio-colors.scss:35]: That's totally fair! Definitely want to make sure it's readable for all. If that means relying more so on the text and radio/checkbox input color, so be it!

**Review by mahoneycm — APPROVED** [review]

**Comment by amyleadem** [comment]
Putting this PR in draft because it is not clear if this will help or harm users with visual processing disorders. Additionally, according to [this comment](https://github.com/uswds/uswds/issues/5492#issuecomment-1737439937) from @mahoneycm, the style change was intentional to meet contrast standards. 

The a11y team is actively engaging with this topic and I recommend we hold off on integrating this change until we can have more discussions about best practice here.

**Comment by amyleadem** [comment]
Opening this PR again after further discussion with the team. We are treating this as a visual regression fix that reduces visual noise. We will follow up with more investigation into how current tile styling affects users with visual processing disorders in issue https://github.com/uswds/uswds/issues/5487.

**Review by thisisdano — APPROVED** [review]
Thanks for all the thought on d discussion on this one. This change is a good step forward, and we'll keep taking more!

**Pull request merged**